compose/foundation/foundation/src/skikoTest/kotlin/androidx/compose/foundation/gestures/SkikoScrollableTest.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.gestures
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.test.ExperimentalTestApi
+import androidx.compose.ui.test.ScrollWheel
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performMouseInput
+import androidx.compose.ui.test.performTouchInput
+import androidx.compose.ui.test.runComposeUiTest
+import androidx.compose.ui.test.swipe
+import androidx.compose.ui.unit.dp
+import kotlin.test.Test
+import kotlin.test.assertTrue
+
+@OptIn(ExperimentalTestApi::class)
+class SkikoScrollableTest {
+    // bug https://github.com/JetBrains/compose-multiplatform/issues/3551 (mouse didn't work)
+    @Test
+    fun recreating_list_state_shouldn_t_break_mouse_scrolling() = runComposeUiTest {
+        var state by mutableStateOf(LazyListState())
+        setContent {
+            LazyColumn(state = state, modifier = Modifier.testTag("list").fillMaxSize()) {
+                items(1000) {
```

Comment:
Sorry, I can't run the tests right now to check:
Does this test reproduce the original issue?
In the original issue it scrolls fine when 100 items added from the start. But it didn't work when first 10 added and then 100 added afterwards. 
And here the test adds 1000 from the beginning. 

Or the items count doesn't matter, because what matters is the LazyListState change?

Reply:
Yes, it reproduces the orignal issue. It was not because we change items, but because we change the state.

I rechecked the original code - it is fixed.

Also, looking at the failed test now